packages/grafana-data/src/types/datasource.ts

```diff
@@ -268,10 +268,20 @@ export abstract class DataSourceApi<
   showContextToggle?(row?: LogRowModel): boolean;
 
   /**
-   * Can be optionally implemented to allow datasource to be a source of annotations for dashboard. To be visible
-   * in the annotation editor `annotations` capability also needs to be enabled in plugin.json.
+   * An annotation processor allows explict control for how annotations are managed.
+   *
+   * It is only necessary to configure an annotation processor if the default behavior is not desirable
+   */
+  annotationProcessor?: AnnotationProcessor<TQuery>;
+
+  /**
+   * Can be optionally implemented to allow datasource to be a source of annotations for dashboard.
+   * This function will only be called if an angular {@link AnnotationsQueryCtrl} is configured and
+   * the {@link annotationProcessor} is undefined
+   *
+   * @deprecated -- prefer using `annotationProcessor`
    */
-  annotationQuery?(options: AnnotationQueryRequest<TAnno>): Promise<AnnotationEvent[]>;
+  annotationQuery?(options: AnnotationQueryRequest<TQuery>): Promise<AnnotationEvent[]>;
```

Comment:
This goes back to the old (pre 7.2) and incorrect types from before.  but that allows us to change the base class without conflicts